### PR TITLE
Nh login redirects

### LINKED ISSUE #65: Add redirects that check if a user isn't logged in
Take a user to a log in page if they aren't logged in when
- [x] they click the new story button
- [x] clicks on an unfinished book

### PR DESCRIPTION
**Pull request by nickhartdev**
#### What's this PR do?
- Adds redirects to the login page from an unfinished book link as well as the start story page
- Refactors the render methods in app to let the redirects work properly
- Adds a toggleHover call on clicking a book
#### Where should the reviewer start?
App.js to see the new redirects
#### How should this be manually tested?
If you click either start story or an unfinished book while not logged in, it should take you to the log in page.
#### Any background context you want to provide?
I added the toggleHover on book click because the hover wouldn't disappear if the redirect fired from an unfinished book click - it would just sit there on top of the log in.
#### What are the relevant tickets?
Resolves #65

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -162,30 +161,30 @@ class App extends Component {
...
-            props.addStory = this.addStory
-            props.author = this.state.currentUser
-            return <StoryEditView {...props} />;
-          }}
-        />
+        <Route exact path="/story-setup">
+          {Object.keys(this.state.currentUser).length === 0 ? (
+            <Redirect to="login" />
+          ) : (
+            <StorySetupView
```
Comment (Abdeboskey) [inline src/components/App/App.js:166]: Niiiiice, sneaky lil ternary

File: src/components/App/App.js [hunk 2]
```diff
@@ -162,30 +161,30 @@ class App extends Component {
...
+              prompts={this.state.prompts}
+              author={this.state.currentUser}
+              addStory={this.addStory}
+              updateStoryData={this.updateStoryData}
+            />
+          )}
+          );
+        </Route>
+        <Route exact path="/story-edit">
+          {Object.keys(this.state.currentUser).length === 0 ? (
```
Comment (Abdeboskey) [inline src/components/App/App.js]: This could also be `!this.state.currentUser.name` just to avoid iterating here!

**Review by Abdeboskey — APPROVED** [review]
Looks great @nickhartdev ! The only request I have is to change these conditions avoid iterating for `Object.keys()`

File: src/components/WelcomePageView/WelcomePageView.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ class WelcomePageView extends Component {
   displayHistory = () => {
     this.setState(prevState => {
       return {
-          showHistory: !prevState.showHistory
+        showHistory: !prevState.showHistory
```
Comment (GreysonElkins) [inline src/components/WelcomePageView/WelcomePageView.js:18]: I'm gonna need to pick your brain on this stuff!

**Review by GreysonElkins — APPROVED** [review]

File: src/components/App/App.js (hunk 2 above)
Comment (nickhartdev) [reply src/components/App/App.js]: Good point! Just updated

File: src/components/WelcomePageView/WelcomePageView.js (hunk 3 above)
Comment (nickhartdev) [reply src/components/WelcomePageView/WelcomePageView.js:18]: I actually didn't add this, I think this came up as a change because I fixed the indentation

Comment (carlymclift) [reply src/components/WelcomePageView/WelcomePageView.js:18]: I actually wrote this, for showing the history

**Review by carlymclift — APPROVED** [review]
Looks great!

**Pull request merged**